main/recording-history.js

```diff
@@ -0,0 +1,314 @@
+};
+
+const getPastRecordings = () => {
+  const recordings = recordingHistory.get('recordings', []);
+  const validRecordings = recordings.filter(({filePath}) => fs.existsSync(filePath));
+  recordingHistory.set('recordings', validRecordings);
+  return validRecordings;
+};
+
+const addRecording = newRecording => {
+  const recordings = [newRecording, ...recordingHistory.get('recordings', [])];
+  const validRecordings = recordings.filter(({filePath}) => fs.existsSync(filePath));
+  recordingHistory.set('recordings', validRecordings);
+  return validRecordings;
+};
+
+const cleanPastRecordings = () => {
+  const recordings = getPastRecordings();
+  for (const recording of recordings) {
+    fs.unlinkSync(recording.filePath);
+  }
+
+  recordingHistory.set('recordings', []);
+};
+
+const cleanUpRecordingPlugins = usedPlugins => {
+  const recordingPlugins = plugins.getRecordingPlugins();
+
+  for (const pluginName of Object.keys(usedPlugins)) {
+    const plugin = recordingPlugins.find(p => p.name === pluginName);
+    for (const [serviceTitle, persistedState] of Object.entries(usedPlugins[pluginName])) {
+      const service = plugin && plugin.recordServices.find(s => s.title === serviceTitle);
+
+      if (service && service.cleanUp) {
+        service.cleanUp(persistedState);
+      }
+    }
+  }
+};
+
+const handleIncompleteRecording = async recording => {
+  cleanUpRecordingPlugins(recording.plugins);
+  try {
+    await execa(ffmpegPath, [
+      '-i',
+      recording.filePath,
+      '-v',
+      'error',
+      '-f',
+      'null',
+      '-'
+    ]);
+
+    return handleRecording(recording);
+  } catch (error) {
+    return handleCorruptRecording(recording, error.stderr);
+  }
+};
+
+const handleRecording = async recording => {
+  addRecording({
+    filePath: recording.filePath,
+    name: recording.name,
+    date: recording.date
+  });
+
+  return showDialog({
+    title: 'Kap didn\'t shut down correctly',
+    detail: 'Looks like Kap crashed during a recording. We were able to locate the file and it appears to be playable.',
+    buttons: [
+      'Close',
+      {
+        label: 'Show in Finder',
+        action: () => shell.showItemInFolder(recording.filePath)
+      },
+      {
+        label: 'Open in Editor',
+        action: () => openEditorWindow(recording.filePath, {recordingName: recording.name})
+      }
+    ]
+  });
+};
+
+const knownErrors = [{
+  test: error => error.includes('moov atom not found'),
+  fix: async filePath => {
+    try {
+      const outputPath = tempy.file({extension: 'mp4'});
+
+      await execa(ffmpegPath, [
+        '-i',
+        filePath,
+        '-vcodec',
+        'copy',
+        '-acodec',
+        'copy',
+        '-movflags',
+        'faststart',
+        outputPath
+      ]);
+
+      return outputPath;
+    } catch {
+
+    }
+  }
+}];
+
+const handleCorruptRecording = async (recording, error) => {
+  const options = {
+    title: 'Kap didn\'t shut down correctly',
```

Comment:
`title` is not shown on macOS. Use `message` instead. And end it with a dot.

Applies in multiple places.

Reply:
This is our custom dialog, so `title` is the title, `detail` is the body, and `message` is an optional extra message towards the bottom.

I'll add the dot